File: May-Leetcoding-Challenge/10. Find the Town Judge/FindTheTownJudge.java
class Solution {
    public int findJudge(int N, int[][] trust) 
    {
        int[] indegree = new int[N];
        int[] outdegree = new int[N];
        
        Arrays.fill(indegree,0,N, 0);
        Arrays.fill(outdegree,0,N, 0);
        
        for(int i=0; i<trust.length ; i++)
        {
            int a = trust[i][0] - 1;
            int b = trust[i][1] - 1;
            
            indegree[b] += 1;
            outdegree[a] += 1;
        }
        
        for(int x=0 ; x<N ; x++)
        {
            if(outdegree[x] == 0 && indegree[x] == N-1)
                return x+1;
        }
        return -1;
    }
}

File: May-Leetcoding-Challenge/10. Find the Town Judge/FindTheTownJudge.py
class Solution:
    def findJudge(self, N: int, trust: List[List[int]]) -> int:
        #  N people labelled from 1 to N
        # GRAPH THEORY
        indegree = [0] * N
        outdegree = [0] * N
        
        for a, b in trust:      #no. of vertices - V
            a -= 1
            b -= 1  #index starts from 0 
            
            outdegree[a] += 1
            indegree[b] += 1
            
        for x in range(N):  #no. of edges - E
            if outdegree[x] == 0 and indegree[x] == N-1:
                return x+1
        ''' total run time - O(V+E) '''
        return -1
            